### ISSUE TITLE
[QST] Implement consistent usage of rapids_cmake_support_conda_env

### ISSUE THREAD

**Issue by vyasr** [rapidsai/rapids-cmake#334]
**What is your question?**

This isn't strictly a question about rapids-cmake so much as a question about its usage across RAPIDS, but there isn't a great place to raise this question so sticking it here for now. Right now it looks like only cuML and raft use this functionality, but I assume we want to handle it consistently in all packages.

**Comment by robertmaynard** [comment]
You mean the python side of cudf correct ? since `libcudf` does call `rapids_cmake_support_conda_env` ( https://github.com/rapidsai/cudf/blob/branch-23.02/cpp/CMakeLists.txt#L127 )

**Comment by vyasr** [comment]
You're right that libcudf uses it, although it does so a bit differently since both cuml and raft both support configuring whether it's called using the `DETECT_CONDA_ENV` option and they also support the environment variable `CONDA_PREFIX`. There's no good reason I can think of why that sort of behavior shouldn't be standardized across all the lib* CMake.

And then yes, figuring out what if anything we should be doing on the Python side.

**Comment by robertmaynard** [comment]
I agree on standardization behavior around conda env detection and behavior.

On that front I am not a fan of the behavior the some projects have where they change the `CMAKE_INSTALL_PREFIX` when using conda
